pkg/jobs/update.go

```diff
@@ -263,6 +266,45 @@ WHERE id = $1
 		}
 	}
 
+	v, err := u.txn.GetSystemSchemaVersion(ctx)
+	if err != nil {
+		return err
+	}
+	if v.AtLeast(clusterversion.V25_1_AddJobsTables.Version()) {
+		if ju.md.Status != "" && ju.md.Status != status {
+			if err := j.StatusStorage().Record(ctx, u.txn, "state", string(ju.md.Status)); err != nil {
+				return err
+			}
+		}
+
+		if progress != nil {
+			var ts hlc.Timestamp
+			if hwm := progress.GetHighWater(); hwm != nil {
+				ts = *hwm
+			}
+
+			if err := j.ProgressStorage().Write(ctx, u.txn, float64(progress.GetFractionCompleted()), ts); err != nil {
+				return err
+			}
+
+			if progress.RunningStatus != beforeProgress.RunningStatus {
+				if err := j.StatusStorage().Set(ctx, u.txn, progress.RunningStatus); err != nil {
+					return err
+				}
+			} else if ju.md.Status != "" && ju.md.Status != status {
+				if err := j.StatusStorage().Clear(ctx, u.txn); err != nil {
```

Comment:
what's the motivation for this branch? I don't follow why if `progress != nil` and the job changed from state without changing running status, we need to clear the status.

Reply:
Anytime the job changes state, we want to clear the status from the old state.

But if we're going to set it to something new in the same txn we don't need to clear it twice. We only know if we're setting it to something new if progress -- where the new thing could be -- is non-nil.

I'll rework to move the Clear to the above case.